packages/gradient-block/src/GradientBlock.tsx

```diff
@@ -0,0 +1,147 @@
+        heightSimple,
+        displayCss,
+    } = blockSettings;
+    const gradientBlockRef = useRef<HTMLDivElement>(null);
+    const [addButtonPositionLeft, setAddButtonPositionLeft] = useState(0);
+    const [currentlyEditingPosition, setCurrentlyEditingPosition] = useState(0);
+    const [showAddButton, setShowAddButton] = useState(false);
+    const [showColorModal, setShowColorModal] = useState(false);
+    const initialColors = !gradientColors || !gradientColors?.length ? DEFAULT_GRADIENT_COLORS : gradientColors;
+    const [colors, setColors] = useState<GradientColor[]>(initialColors);
+
+    const gradientOrientation = isOrientationCustom
+        ? orientationCustom
+        : gradientOrientationValues[orientationSimple ?? DEFAULT_ORIENTATION_VALUE];
+
+    const gradientBlockHeight = isHeightCustom
+        ? heightCustom
+        : gradientHeightValues[heightSimple ?? DEFAULT_HEIGHT_VALUE];
+
+    const parseGradientColorsToCss = () => {
+        let colorsAsString = '';
+        for (const color of colors.sort((a, b) => a.position - b.position)) {
+            colorsAsString += `, ${toHexString(color.color)} ${color.position}%`;
+        }
+
+        return `linear-gradient(${gradientOrientation}deg${colorsAsString})`;
+    };
+
+    const handleMouseMove = (event: MouseEvent<HTMLDivElement>) => {
+        if (showColorModal) {
+            return;
+        }
+        const rect = event.currentTarget.getBoundingClientRect();
+        const { clientX } = event.nativeEvent;
+        const relativeMouseX = clientX - rect.x;
+        setAddButtonPositionLeft(relativeMouseX - 16 / 2);
+        setShowAddButton(true);
+    };
+
+    useEffect(() => {
+        if (!gradientColors || !gradientColors?.length) {
+            setBlockSettings({
+                gradientColors: DEFAULT_GRADIENT_COLORS,
+            });
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [gradientColors]);
+
+    useEffect(() => {
+        setBlockSettings({
+            gradientColors: colors,
+        });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [gradientOrientation, colors]);
+
+    const cssValue = parseGradientColorsToCss();
+
+    return (
+        <div data-test-id="gradient-block" ref={gradientBlockRef}>
+            <div className="tw-border tw-border-line-strong tw-rounded tw-p-0.5">
+                <div
+                    data-test-id="gradient-block-display"
+                    className="tw-w-full tw-h-4 tw-rounded"
+                    style={{
+                        height: gradientBlockHeight,
+                        background: cssValue,
+                    }}
+                />
+            </div>
+            {isEditing ? (
+                <div>
+                    <div className="tw-relative">
+                        <div
+                            data-test-id="gradient-block-divider"
+                            onMouseOver={handleMouseMove}
```

Comment:
```suggestion
                            onMouseMove={handleMouseMove}
```

results in better performance for me